yetanotherco/aligned_layer
# feat: operators version service

Add operator version tracker

Exposes two endpoints:

- POST to /versions for operator to post its version on startup, verifies operator signature
- GET to /versions to list operator versions

# To test

Start anvil and aggregator

Start operator tracker: `make tracker_devnet_start`

Register operator: `make operator_full_registration`
Build operator: `make build_operator`
Start operator: `./operator/build/aligned-operator start --config ./config-files/config.yaml`

> [!WARNING]  
> Note that you can't use `make operator_start` because operator version is not included in executable if not built.

You can then get operator versions by running `curl http://localhost:3000/versions`

You can test an update by changing OPERATOR_VERSION in makefile, rebuilding and restarting operator
